src/adaptors/http2/http2_adaptor.c

```diff
@@ -1966,11 +1966,25 @@ uint64_t handle_outgoing_http(qdr_http2_stream_data_t *stream_data)
                 SET_ATOMIC_FLAG(&conn->delay_buffer_write);
             }
 
-            stream_data->stream_id =
-                nghttp2_submit_headers(conn->session, flags, stream_id, NULL, hdrs, actual_count, stream_data);
-
-            if (stream_id != -1) {
-                stream_data->stream_id = stream_id;
+            int32_t ret_val = nghttp2_submit_headers(conn->session, flags, stream_id, NULL, hdrs, actual_count, stream_data);
+
+            // The call to nghttp2_submit_headers can return 3 possible values.
+            // 1. The new stream id > 0 if this is a request and the passed in stream id was -1.
+            // 2. zero if this is response and there was no error when submitting the response headers
+            // 3. returns an nghttp2 specific error code less than zero if there was some error calling nghttp2_submit_headers.
+            if (ret_val < 0) {
+                // An error code was returned by nghttp2 when calling nghttp2_submit_headers. This was a failure in submitting the headers
+                // Log the failure code returned by nghttp2 and do not proceed
+                qd_log(http2_adaptor->protocol_log_source, QD_LOG_ERROR, "[C%"PRIu64"] nghttp2_submit_headers failed, ret_val=%"PRId32", closing connection", conn->conn_id, ret_val);
```

Comment:
Can this error code be stored in a vflow log record?

Reply:
I added the vflow records. Please take a look. Thanks.